front/app/containers/Admin/projects/project/files/components/FilesUpload/index.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useState } from 'react';
+
+import {
+  Box,
+  CheckboxWithLabel,
+  Text,
+  Title,
+} from '@citizenlab/cl2-component-library';
+import { useDropzone } from 'react-dropzone';
+import { useParams } from 'react-router-dom';
+
+import { useIntl } from 'utils/cl-intl';
+
+import messages from '../messages';
+
+import FileDropzone from './components/FileDropzone';
+import FileUploadActions from './components/FileUploadActions';
+import SelectedFile from './components/SelectedFile';
+import { FileWithMeta, UploadStatus } from './types';
+
+type Props = {
+  setModalOpen: (open: boolean) => void;
+};
+
+const MAX_FILE_SIZE = 50 * 1024 * 1024;
+const MAX_FILES = 35;
+
+const FilesUpload = ({ setModalOpen }: Props) => {
+  const { formatMessage } = useIntl();
+  const { projectId } = useParams() as { projectId: string };
+
+  const [fileList, setFileList] = useState<FileWithMeta[]>([]);
+  const [hasStartedUploading, setHasStartedUploading] = useState(false);
+  const [showMaxNumberFilesMessage, setShowMaxNumberFilesMessage] =
+    useState(false);
+
+  // Create a React dropzone with the specified options
+  const { getRootProps, getInputProps, open } = useDropzone({
```

Comment:
Can we rename these methods `getRootProps` and `getInputProps` to make it clear that they refer to the `useDropzone` hook in some way. Just looking at the name sounds a bit confusing, especially when they get passed down as props so in the other component it's not clear what they are in reference to.